packages/maplibre/lib/helpers/map.helpers.ts

```diff
@@ -0,0 +1,59 @@
+import { Map, StyleSpecification } from "maplibre-gl";
+import {
+  Dataset,
+  LayerContextWithStyle,
+  LayerSpecificationWithSource,
+} from "../maplibre.models";
+import { FeatureCollection, Geometry } from "geojson";
+import { contextStyleToMaplibreLayers } from "./style.helpers";
+
+/**
+ * Remove all layers from a given source in the map.
+ * @param map
+ * @param sourceId
+ */
+export function removeLayersFromSource(map: Map, sourceId: string) {
+  const layers = map.getStyle().layers;
+  const layersWithSource = layers.filter(
+    (layer) => layer.type !== "background",
+  ) as LayerSpecificationWithSource[];
+  const layerIds = layersWithSource
+    .filter(
+      (layer) => layer.hasOwnProperty("source") && layer.source === sourceId,
+    )
+    .map((layer) => layer.id);
+  layerIds.forEach((layer) => map.removeLayer(layer));
+}
+
+/**
+ * Create a Maplibre source and layers from a GeoJSON MapContextLayer and its style.
+ * @param layerModel
+ * @param geojson
+ */
+export function createDatasetFromGeoJsonLayer(
+  layerModel: LayerContextWithStyle,
+  geojson: FeatureCollection<Geometry | null> | string,
+): Dataset {
+  const sourceId =
+    layerModel.id || Math.floor(Math.random() * 1000000).toString();
```

Comment:
```suggestion
    layerModel.id ?? Math.floor(Math.random() * 1000000).toString();
```

In case the `id` value is falsy but not undefined